src/@next/components/atoms/Select/Select.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useRef, useState } from "react";
+
+import { Icon, Input } from "..";
+import { useHandlerWhenClickedOutside } from "../../../hooks";
+
+import { SelectOptionsList } from "./SelectOptionsList";
+import * as S from "./styles";
+import { IProps } from "./types";
+
+const filterOptions = (
+  searchValue: string,
+  options: string[],
+  shouldFilter: boolean
+) => {
+  if (shouldFilter && searchValue) {
+    return options.filter(element => {
+      return element.toLowerCase().includes(searchValue.toLowerCase())
+        ? element
+        : "";
+    });
+  } else {
+    return options;
+  }
+};
+
+export const Select: React.FC<IProps> = ({
+  options,
+  defaultValue,
+  label,
+}: IProps) => {
+  const [isOpen, setOpen] = useState(false);
+  const [value, setValue] = useState(defaultValue || "");
+  const [searchValue, setSearchValue] = useState("");
+  const { setElementRef } = useHandlerWhenClickedOutside(() => {
+    setOpen(false);
+  });
+
+  const activeOptionRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (activeOptionRef.current && isOpen) {
+      activeOptionRef.current.scrollIntoView();
+      activeOptionRef.current.focus();
+    }
+  }, [isOpen]);
+
+  const saveValue = (option: string) => {
+    setOpen(false);
+    setValue(option);
+  };
+
+  const shouldFilter = value !== searchValue;
+  const filteredOptions = filterOptions(searchValue, options, shouldFilter);
+
+  return (
+    <S.Wrapper
+      onClick={() => {
+        setOpen(!isOpen);
+        setSearchValue(value);
```

Comment:
Setting search value equal to current value here is bad UX. I'd get rid off it completely and use Input as follows:
```
<Input
  placeholder={value}
  value={isOpen ? searchValue : value}
```
This way, before a user inputs anything he still gets the original `value` but he starts from an empty string